pkg/apis/storage/types.go

```diff
@@ -203,6 +227,10 @@ type VolumeError struct {
 	// information.
 	// +optional
 	Message string
+
+	// errorCode specifies the type of error encountered during Attach or Detach operation.
+	// +optional
+	ErrorCode *VolumeErrorCode
```

Comment:
We're missing some validation logic on the enum.

Reply:
Changed field type, please let me know if there is any additional validation to add.